### PR TITLE
Add invoice link to transaction form

### PR DESCRIPTION
**Pull request by dissident**

### CODE REVIEW

**Comment by dissident** [comment]
https://trello.com/c/zZVD6075/339--

File: spec/features/update_transaction_spec.rb [hunk 1]
```diff
@@ -99,4 +99,37 @@
...
+
+    before do
+      visit root_path
+      find("##{dom_id(transaction)} .comment").click
+      page.has_css?("##{dom_id(transaction, :edit)}")
+      click_on 'Update'
+    end
+
+    it "not view link to invoice after the fields" do
+      expect(subject).not_to have_link("Created from invoice")
```
Comment (AlexanderUlitin) [inline spec/features/update_transaction_spec.rb]: @dissident we usually use `.to_not` method, please fix.

File: spec/features/update_transaction_spec.rb [hunk 2]
```diff
@@ -99,4 +99,37 @@
...
+
+    before do
+      visit root_path
+      find("##{dom_id(transaction)} .comment").click
+      page.has_css?("##{dom_id(transaction, :edit)}")
+      click_on 'Update'
+    end
+
+    it "view link to invoice after the fields" do
+      expect(subject).to have_link("Created from invoice")
```
Comment (AlexanderUlitin) [inline spec/features/update_transaction_spec.rb]: @dissident please check href for this link too

File: spec/features/update_transaction_spec.rb (hunk 1 above)
Comment (shir) [inline spec/features/update_transaction_spec.rb]: @AlexanderUlitin You are wrong. `.not_to` is better to read.

File: app/views/transactions/_form.html.slim [hunk 3]
```diff
@@ -63,4 +63,7 @@
   = f.input :date, as: :string, placeholder: 'Date',
     input_html: { class: 'datepicker', value: f.object.date.present? ? l(f.object.date.to_date) : nil }
 
+  - if @invoice.present? || f.object.invoice_id
+    = link_to "Created from invoice", @invoice, class: 'pull-right'
```
Comment (AlexanderUlitin) [inline app/views/transactions/_form.html.slim]: @dissident I'm not sure if we have `@invoice` on edit existing transaction. Please check it and fix. We don't need to show link to Invoice on create form, only when edit existing transactions.

File: spec/features/update_transaction_spec.rb (hunk 1 above)
Comment (AlexanderUlitin) [inline spec/features/update_transaction_spec.rb]: Ok, discussed with Shir, let's write as you feel more comfortable to read.

File: spec/features/update_transaction_spec.rb (hunk 2 above)
Comment (dissident) [inline spec/features/update_transaction_spec.rb]: done

File: app/views/transactions/_form.html.slim (hunk 3 above)
Comment (dissident) [inline app/views/transactions/_form.html.slim]: You're right, we have not @invoice on edit transaction action, I do not understand why this code to work well.

**Pull request merged**